### PR TITLE
gh-1469 - added sample elements for split points operation

### PR DESCRIPTION
**Pull request by p013570**
There is a bit of extra complexity in the handler as I have attempted to make an abstract class that is generic so that it is easy for different Stores to implement it. The operation SampleElementsForSplitPoints returns a List<T> where T is the split point type. For Accumulo and HBase this is a String, but other stores may use a different type. For Accumulo and HBase the split points are Base64 encoded.

### CODE REVIEW

File: core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/AbstractSampleElementsForSplitPointsHandler.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import java.util.stream.Stream;
+
+public abstract class AbstractSampleElementsForSplitPointsHandler<T, S extends Store> implements OutputOperationHandler<SampleElementsForSplitPoints<T>, List<T>> {
+    public static final int MAX_SAMPLED_ELEMENTS_DEFAULT = 10000000;
+
+    private int maxSampledElements = MAX_SAMPLED_ELEMENTS_DEFAULT;
+
+    @Override
+    public List<T> doOperation(final SampleElementsForSplitPoints<T> operation, final Context context, final Store store) throws OperationException {
+        S typedStore = (S) store;
```
Comment (gaffer01) [inline core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/AbstractSampleElementsForSplitPointsHandler.java]: Should be final I think

File: store-implementation/hbase-store/src/main/java/uk/gov/gchq/gaffer/hbasestore/operation/handler/SampleElementsForSplitPointsHandler.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            numSplits = getNumAccumuloSplits(store);
+        }
+        return numSplits;
+    }
+
+    private int getNumAccumuloSplits(final HBaseStore store) {
+        int numberRegions;
+        try {
+            numberRegions = store.getConnection().getAdmin().getTableRegions(store.getTableName()).size();
+            LOGGER.debug("Number of tablet servers is {}", numberRegions);
```
Comment (gaffer01) [inline store-implementation/hbase-store/src/main/java/uk/gov/gchq/gaffer/hbasestore/operation/handler/SampleElementsForSplitPointsHandler.java]: Pedantic but should be "region servers" not "tablet servers"

**Comment by gaffer01** [comment]
I've made a couple of minor comments.

Main comment is how are the list of split points applied to a table? There is a `SplitStore` operation but that takes a path to a file of split points, whereas the new operation added here outputs a list of split points. Is the intention that the split points will be written to file and then read by the `SplitStore` operation? Should we add the ability to split a store directly from a list of split points?

**Comment by p013570** [comment]
@gaffer01 - added made your suggested changes. I've deprecated SplitStore and created SplitStoreFromFile and SplitStoreFromIterable. The SplitStoreFromFile reads the file locally or in HDFS then delegates to SplitStoreFromIterable.

**Pull request merged**